### DISCUSSION TITLE
How to execute migrations without a database

### DISCUSSION THREAD

**Discussion by lukewis** [fluentmigrator/fluentmigrator#1866] [Q&A]
I have some json configuration files whose schema may change from version to version of our application.  I'd like to use fluent migrations to migrate these files, but since it's editing files in a git repo, there is no "database" per se.  I know I could use SQLite, but putting a database into a git repo feels wrong.  Is there a way to write a custom provider (like write version history to a JSON file, etc)?

**Comment by PhenX** [comment]
I don't think Fluent Migrator is the right tool for your usage, JSON schema and database schema are not the same, maybe something like https://www.npmjs.com/package/@eidellev/json-migrate would be better.
